docs/tools/firmware.sh

```diff
@@ -517,101 +806,96 @@ case "$os" in
+				sudo mount ${verbose} /dev/nvme0n1p1 "$mountpoint"
+				sudo tar --warning=no-unknown-keyword ${verbose} -xC "${workdir}" -f "$mountpoint/firmware-raw.tar.gz"
+				sudo chown -R "$USER" "${workdir}"
+				create_firmware_archive "${workdir}" "${workdir}/firmware-renamed.tar" ${verbose}
 
-				sudo tar ${verbose} -xC /lib/firmware/brcm -f ${workdir}/firmware-renamed.tar
+				sudo tar ${verbose} -xC /lib/firmware/brcm -f "${workdir}/firmware-renamed.tar"
 
-				for file in "$mountpoint/brcmfmac4364b2-pcie.txt" \
-					    "$mountpoint/brcmfmac4364b2-pcie.txcap_blob"
+				for file in "$workdir/brcmfmac4364b2-pcie.txt" \
+					"$workdir/brcmfmac4364b2-pcie.txcap_blob"
 				do
-					if [ -f "$file" ]
-					then
-						sudo cp ${verbose} $file /lib/firmware/brcm
+					if [ -f "$file" ]; then
+						sudo cp ${verbose} "$file" /lib/firmware/brcm
 					fi
 				done
-				echo "Reloading Wi-Fi and Bluetooth drivers"
-				sudo modprobe -r ${verbose} brcmfmac_wcc || true
-				sudo modprobe -r ${verbose} brcmfmac || true
-				sudo modprobe ${verbose} brcmfmac || true
-				sudo modprobe -r ${verbose} hci_bcm4377 || true
-				sudo modprobe ${verbose} hci_bcm4377 || true
+				reload_kernel_modules
 				echo -e "\nKeeping a copy of the firmware and the script in the EFI partition shall allow you to set up Wi-Fi again in the future by running this script or the commands told in the macOS step in Linux only, without the macOS step."
-				read -p "Do you want to keep a copy? (y/N)" input
+				read -rp "Do you want to keep a copy? (y/N)" input
 				if [[ ($input != y) && ($input != Y) ]]
 				then
 					echo -e "\nRemoving the copy from the EFI partition"
-					for file in "$mountpoint/brcmfmac4364b2-pcie.txt" \
-					            "$mountpoint/brcmfmac4364b2-pcie.txcap_blob" \
-					            "$mountpoint/firmware.tar.gz" \
+					for file in "$mountpoint/firmware-raw.tar.gz" \
 					            "$mountpoint/firmware.sh"
 					do
 						if [ -f "$file" ]
 						then
-							sudo rm ${verbose} $file
+							sudo rm ${verbose} "$file"
 						fi
 					done
 				fi
-				sudo rm -r ${verbose} ${workdir}
-				sudo umount $mountpoint
-				sudo rmdir $mountpoint
+				sudo rm -r ${verbose} "${workdir}"
+				sudo umount "$mountpoint"
+				sudo rmdir "$mountpoint"
 				echo -e "\nDone!"
 				;;
 			(2)
 				echo -e "\nChecking for missing dependencies"
 				# Load the apfs driver, and install if missing
-				sudo modprobe ${verbose} apfs 2>/dev/null || apfs_install
+				sudo modprobe ${verbose} apfs 2>/dev/null || install_package linux-apfs-rw
 				unmount_macos_and_cleanup () {
-					sudo rm -r ${verbose} ${workdir} || true
+					sudo rm -r ${verbose} "${workdir}" || true
 					for i in 0 1 2 3 4 5
 					do
-						if [[ ${verbose} = -v ]]
-						then
-							sudo umount -v ${macosdir}/vol${i} || true
-						else
-							sudo umount ${macosdir}/vol${i} 2>/dev/null || true
-						fi
+						sudo umount "${macosdir}/vol${i}" 2>&1 | log || true
```

Comment:
Will this change mimic umount -v if run with -v

Reply:
I need to pass`${verbose}` to umount. the `| log` just ignores whatever is passed to it in normal mode and outputs to stdout in verbose mode.

Reply:
Ok